packages/@divvi/mobile/src/earn/EarnDepositConfirmationScreen.test.tsx

```diff
@@ -428,9 +430,40 @@ describe('EarnDepositConfirmationScreen', () => {
           'reviewTransaction.totalPlusFees'
         )
         expect(getByTestId('TotalInfoBottomSheet/Total/Value')).toHaveTextContent(totalFeesValue)
+
+        // footer with disclaimer and confirm button
+        expect(
+          getByText('earnFlow.depositConfirmation.disclaimer, {"providerName":"Aave"}')
+        ).toBeTruthy()
+        expect(getByTestId('EarnDepositConfirmation/TermsAndConditions')).toBeTruthy()
+        expect(getByTestId('EarnDepositConfirmation/ConfirmButton')).toHaveTextContent('deposit')
       })
 
-      it('pressing cancel fires analytics event', () => {
+      it('renders different disclaimer when there is no terms url', () => {
+        const { getByTestId, getByText } = render(
+          <Provider store={createMockStore({ tokens: { tokenBalances: mockTokenBalances } })}>
+            <EarnDepositConfirmationScreen
+              {...getMockStackScreenProps(Screens.EarnDepositConfirmationScreen, {
+                ...props,
+                pool: {
+                  ...props.pool,
+                  dataProps: { ...props.pool.dataProps, termsUrl: undefined },
+                },
+              })}
+            />
+          </Provider>
+        )
+
+        expect(
+          getByText(
+            'earnFlow.depositConfirmation.noTermsUrlDisclaimer, {"appName":"Test App","providerName":"Aave"}'
+          )
+        ).toBeTruthy()
+        expect(getByTestId('EarnDepositConfirmation/ProviderDocuments')).toBeTruthy()
+        expect(getByTestId('EarnDepositConfirmation/AppTermsAndConditions')).toBeTruthy()
+      })
+
+      it('pressing back button fires analytics event', () => {
```

Comment:
So it reads as English with the starting "it"
```suggestion
      it('triggers an analytics event when the back button is pressed', () =>
```